src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -464,10 +472,9 @@ protected async Task<ConnectNewHeadersResult> PresentHeadersLockedAsync(List<Blo
                 }
                 else
                 {
-                    this.cachedHeaders.Clear();
-
                     // Resync in case we can't connect the header.
-                    await this.ResyncAsync().ConfigureAwait(false);
+                    this.cachedHeaders.Clear();
+                    this.ResyncAsync();
```

Comment:
await

Reply:
or change resync async to return void since its not really async

Reply:
Ah I missed this one. Will fix, Yeah we are going to cleanup the async methods in another PR ;)